src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -227,7 +227,15 @@ void ContinuousBatchingPipeline::ContinuousBatchingImpl::initialize_pipeline(
             std::make_shared<ModelRunner>(infer_request, m_scheduler->get_block_size(), m_num_decoder_layers);
     }
 
-    m_sampler = std::make_shared<Sampler>(m_tokenizer);
+    size_t sampler_num_threads = std::thread::hardware_concurrency(); // default value for CB
+    // Extract sampler_num_threads value from properties map
+    auto sampler_num_threads_it = properties.find("sampler_num_threads");
```

Comment:
such properties are passed to plugin's `compile_model` and plugin will fail with unknown property.

You need to drop this value from `properties` before it's passed to plugin.

Reply:
Secondly, pass `sampler_num_threads` from CB adapter as `1`